compiler/util/src/org/jetbrains/kotlin/utils/SmartIdentityTable.kt

```diff
@@ -80,7 +101,15 @@ class SmartIdentityTable<K, V> {
         }
     }
 
+    fun putAll(other: SmartIdentityTable<K, V>) {
+        for (key in other.keys) {
+            this[key] = other[key]!!
+        }
+    }
+
     companion object {
         private const val ARRAY_UNTIL_SIZE = 10
     }
+
+    data class Entry<K, V>(val key: K, val value: V)
```

Comment:
Might be nice to inherit this from `Map.Entry<K, V>` and make `entries` return `Sequence<Map.Entry<K, V>>` to avoid rewrapping map entries into these entries in the implementation